design/api.md

```diff
@@ -11,91 +11,85 @@ This spec describes
 
 Today we use a REST API. In the future we may switch to an RPC-based design. When/if this happens mostly depends on a plan for implementing the UCAN invocation spec.
 
+| Method | Path | Comment |
+|--------|------|---------|
+| POST | [`/api/v0/auth/email/verify`](#post-apiv0authemailverify) | Trigger email verification |
+| POST | [`/api/v0/account`](#post-apiv0account) | Create a full account |
+| GET | [`/api/v0/account/:did`](#get-apiv0accountdid) | Get account information by DID |
+| GET | [`/api/v0/account/:username/did`](#get-apiv0accountusernamedid) | Get the did of an account by username |
+| PUT | [`/api/v0/account/:did/username/:username`](#put-apiv0accountdidusernameusername) | Change an account's username |
```

Comment:
PUT is generallt meant for inserting the entire record. Have you considered PATCH? e.g.

```
PATCH /api/v0/account/:did?username=newname
```

Reply:
Yeah, I didn't. Should be PATCH probably.

Reply:
Also, I should probably change the username to be passed via the UCAN facts.
And maybe even get rid of the did fragment

Reply:
Ah so that they're signed over? I would personally wait for Invocations to land. We understand the tradeoff right now, I don't know if we need to invent stuff here when full solutions are immenent.